src/gui/filedetails/sharemodel.h

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2022 by Claudio Cambra <claudio.cambra@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef OCC_SHAREMODEL_H
+#define OCC_SHAREMODEL_H
+
+#include "accountstate.h"
+#include "folder.h"
+#include "sharemanager.h"
+#include "sharepermissions.h"
+
+#include <QAbstractListModel>
+#include <QString>
+#include <QDialog>
```

Comment:
@claucambra Is this include needed? Could you also go through all the includes and remove the leftvoers that are not in use anymore?

Reply:
Removed all unused includes now, thanks for pointing out